.github/workflows/ci.yml

```diff
@@ -12,6 +12,7 @@ on:
 jobs:
   ci:
     runs-on: ubuntu-latest
+    if: github.repository == 'rust-lang/rustc-dev-guide'
     env:
```

Comment:
nit: For this and for `date-check.yml` I think it would be better to have the `if` before `runs-on` because right now it's in between `runs-on` and `env` which are in the same category.